src/differentiation/differentiation.jl

```diff
@@ -47,98 +49,119 @@ for example [`gradient`](@ref Manifolds.gradient(::AbstractManifold, ::Any, ::An
+    return copyto!(X, _jacobian(f, p, backend))
+end
+
 """
     CurrentDiffBackend(backend::AbstractDiffBackend)
 
 A mutable struct for storing the current differentiation backend in a global
-constant [`_current_diff_backend`](@ref).
+constant [`_current_default_differential_backend`](@ref).
 
 # See also
 
-[`AbstractDiffBackend`](@ref), [`diff_backend`](@ref), [`diff_backend!`](@ref)
+[`AbstractDiffBackend`](@ref), [`default_differential_backend`](@ref), [`set_default_differential_backend`](@ref)
 """
 mutable struct CurrentDiffBackend
     backend::AbstractDiffBackend
 end
 
 """
-    _current_diff_backend
+    _current_default_differential_backend
 
 The instance of [`Manifolds.CurrentDiffBackend`](@ref) that stores the globally default
 differentiation backend.
 """
-const _current_diff_backend = CurrentDiffBackend(NoneDiffBackend())
-
+const _current_default_differential_backend = CurrentDiffBackend(NoneDiffBackend())
 """
-    _diff_backends
+    default_differential_backend() -> AbstractDiffBackend
 
-A vector of valid [`Manifolds.AbstractDiffBackend`](@ref).
+Get the default differentiation backend.
 """
-const _diff_backends = AbstractDiffBackend[]
+default_differential_backend() = _current_default_differential_backend.backend
 
 """
-    diff_backend() -> AbstractDiffBackend
-
-Get the current differentiation backend.
-"""
-diff_backend() = _current_diff_backend.backend
-
-"""
-    diff_backend!(backend::AbstractDiffBackend)
+    set_default_differential_backend!(backend::AbstractDiffBackend)
 
 Set current backend for differentiation to `backend`.
 """
-function diff_backend!(backend::AbstractDiffBackend)
-    _current_diff_backend.backend = backend
+function set_default_differential_backend!(backend::AbstractDiffBackend)
+    _current_default_differential_backend.backend = backend
     return backend
 end
 
-"""
-    diff_backends() -> Vector{AbstractDiffBackend}
-
-Get vector of currently valid differentiation backends.
-"""
-diff_backends() = _diff_backends
-
-_derivative(f, t) = _derivative(f, t, diff_backend())
-
-_derivative!(f, X, t) = _derivative!(f, X, t, diff_backend())
+@doc raw"""
+    ODEExponentialRetraction{T<:AbstractRetractionMethod, B<: AbstractBasis} <: AbstractRetractionMethod
 
-_gradient(f, p) = _gradient(f, p, diff_backend())
+This retraction approximates the exponential map by solving the correspondig ODE.
+Let ``p\in\mathal M`` and ``X\in T_p\mathcal M`` denote the input for the exponential map
+and ``d`` denote the [`manifold_dimension`](@ref) of `M`.
 
-_gradient!(f, X, p) = _gradient!(f, X, p, diff_backend())
+This the ODE is formulated in a chart constructed using an [`AbstractBasis`](@ref) `B` and
+and using either an embedding or a [`AbstractRetractionMethod`](@ref) `R` as follows.
+Given some coordinates ``c\in ℝ^d`` - these can be used to form a tangent vector
+with restect to th basis `B, i.e. ``c \mapsto Y=``[`get_vector`](@ref)`(M, p, c, B)`.
```

Comment:
I'd prefer to leave the details regarding how the basis is used out of this PR -- this still needs to be worked out for cases other than the induced basis of a chart.

Reply:
I must have missed those lines in my interactive merge (removing parts from the last PR). Will update this, sure.